### DISCUSSION TITLE
minified sources?

### DISCUSSION THREAD

**Discussion by ZeeD** [rokucommunity/brighterscript#750] [General]
Is there some way to ask `bsc` to generate minified .brs files? like putting every source/*.bs file in a single .brs, do some name mangling, etc...

I have this requirement to fit my library in a very small space...

**Comment by TwitchBronBron** [comment]
I'm not aware of any existing ways to do this. 

The new file api coming in #408 (hopefully will land this week or next) will be adding a full file transformation pipeline that would make it easy for a plugin to implement something like this. 

However, I'm not sure mangling/minifying/merging would actually make a significant difference in the final file size. The .zip and squashfs processes are extremely efficient, especially on text-based files. I would suggest you do do some tests to see what the real-world zip size reduction would be for your app.

As a quick example, I took the `source` and `components` folders from the Fubo app. I intentionally excluded any images or assets, as those binary files generally don't compress as nicely as text. Then I ran a very rough mangler over all the source files. The mangle code does NOT produce valid output, but it gives us a best-case size of mangled output. Here's the code I used to mangle the project: 

<details>
  <summary>View mangle.ts</summary>

```typescript
import * as fastGlob from 'fast-glob';
import * as fsExtra from 'fs-extra';
const files = fastGlob.sync('**/*', {
    cwd: 'src',
    absolute: false
});
for (const file of files) {
    let contents = fsExtra.readFileSync(`src/${file}`).toString()
    contents = contents
        //remove comments
        .replace(/'.*(?=\n|$)/g, '')
        //condense consecutive newlines to a single newline
        .replace(/\r?\n+/g, '\n')
        //mangle all words
        .replace(/(?<!")(\w+)/g, function (match, word) {
            //skip the known brightscript keywords
            if ([
                'function', 'sub', 'if', 'then', 'else', 'end', 'while', 'true', 'false', 'continue', 'exit', 'not', 'as', 'return', 'each',
                'invalid', 'integer', 'longinteger', 'dynamic', 'boolean', 'float', 'object', 'string',
                'createchild', 'createobject', 'update', 'type',
            ].includes(word.toLowerCase())) {
                return word;
            } else {
                return word[0];
            }
        })
        //condense whitespace
        .replace(/( |\t)+/g, ' ')
        //trim every line
        .replace(/^.*$/gm, function (match) {
            return match.trim();
        });
    fsExtra.outputFileSync(`./dest/${file}`, contents);
}
```
</details>

Then, here's the sizes of each.
![image](https://user-images.githubusercontent.com/2544493/204813517-6201f9b4-1fc7-4c27-baf8-bb6c3963a897.png)

The original code zipped is 965KB, and the mangled zip is 583KB zipped.

**Reply by ZeeD** [reply]
I totally agree with you about the fact that this shouldn't really be impactful on the .zip file. Unfortunately in my specific case I have received a very strict requirement about the source code itself - not really sure why, honestly.

apart from that, I may look at the new plugin api, when they will be available
